SDWebImage/Core/SDAnimatedImageView.m

```diff
@@ -246,11 +243,8 @@ - (void)setImage:(UIImage *)image
         if (self.shouldAnimate) {
             [self startAnimating];
         }
-        
-        [self.layer setNeedsDisplay];
-#if SD_MAC
-        [self.layer displayIfNeeded]; // macOS's imageViewLayer may not equal to self.layer. But `[super setImage:]` will impliedly mark it needsDisplay. We call `[self.layer displayIfNeeded]` to immediately refresh the imageViewLayer to avoid flashing
```

Comment:
This `displayIfNeeded` is useless for new solution, since I don't care the `self.layer`'s display, the actually rendering is happended on `self.imageViewLayer`